File: FixSkkuFront/app/src/main/java/com/skku/fixskkufront/MainActivity.java
package com.skku.fixskkufront;


import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.os.Bundle;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;
import java.util.ArrayList;
import java.util.List;


public class MainActivity extends AppCompatActivity {
    RecyclerView recyclerView;
    TextView welcomeTextView;
    EditText messageEditText;
    ImageButton sendButton;
    ImageButton backButton;
    List<Message> messageList;
    MessageAdapter messageAdapter;
    ImageView imageView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        messageList = new ArrayList<>();

        recyclerView = findViewById(R.id.recycler_view);
        welcomeTextView = findViewById(R.id.welcome_text);
        messageEditText = findViewById(R.id.message_edit_text);
        sendButton = findViewById(R.id.send_btn);
        imageView = findViewById(R.id.SKKU_imageView);
        backButton = findViewById(R.id.back_btn);


        //setup recycler view
        messageAdapter = new MessageAdapter(messageList);
        recyclerView.setAdapter(messageAdapter);
        LinearLayoutManager llm = new LinearLayoutManager(this);
        //llm.setStackFromEnd(true); //메세지가 밑에서부터 쌓이게 하는 코드
        recyclerView.setLayoutManager(llm);

        /* 아이콘 넣으려고 했으나, 실패했습니다. ㅠ
        * 첫 메세지 */
        addToChat("안녕하세요! 픽스꾸 봇 입니다! 무엇을 도와드릴까요?",Message.SENT_BY_BOT);

        /* 메세지 전송 */
        sendButton.setOnClickListener((v)->{
            String question = messageEditText.getText().toString().trim();
            addToChat(question,Message.SENT_BY_ME); // 메세지 보내기
            messageEditText.setText("");
            //welcomeTextView.setVisibility(View.GONE);
            imageView.setAlpha(0.2f);  // 투명도를 20%로 설정

        });
        /* 뒤로가기 */
        backButton.setOnClickListener((v)->{
            addToChat("Temp Answer...",Message.SENT_BY_BOT); // 메세지 답변
        });
    }

    void addToChat(String message,String sentBy){
        runOnUiThread(new Runnable() {
            @Override
            public void run() {
                messageList.add(new Message(message,sentBy));
                messageAdapter.notifyDataSetChanged();
                recyclerView.smoothScrollToPosition(messageAdapter.getItemCount());
            }
        });
    }


}

File: FixSkkuFront/app/src/main/java/com/skku/fixskkufront/MainActivity.kt
package com.skku.fixskkufront

import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.text.InputType
import android.view.View
import android.view.WindowInsets
import android.view.WindowInsetsController
import android.widget.Button
import android.widget.EditText
import android.widget.FrameLayout
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        hideSystemUI()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        val userButton: Button = findViewById<Button>(R.id.userBtn)
        val adminButton: Button = findViewById<Button>(R.id.adminBtn)

        userButton.setOnClickListener {
            val intent = Intent(this@MainActivity, UserActivity::class.java)
            startActivity(intent)
        }

        adminButton.setOnClickListener {
            showAdminDialog()
        }
    }

    private fun showAdminDialog() {
        val builder = AlertDialog.Builder(this)
        builder.setTitle("Enter Admin Code")

        // Create a container to hold the EditText
        val container = FrameLayout(this)
        val input = EditText(this)
        input.inputType = InputType.TYPE_CLASS_NUMBER

        // Set layout parameters to match the width of the dialog title and buttons
        val params = FrameLayout.LayoutParams(
            FrameLayout.LayoutParams.MATCH_PARENT,
            FrameLayout.LayoutParams.WRAP_CONTENT
        )
        params.setMargins(50, 0, 50, 0)  // Set margin to add some space
        input.layoutParams = params

        container.addView(input)
        builder.setView(container)

        builder.setPositiveButton("Enter") { dialog, _ ->
            val enteredCode = input.text.toString()
            if (enteredCode == "1234") {
                val intent = Intent(this@MainActivity, AdminActivity::class.java)
                startActivity(intent)
            } else {
                Toast.makeText(this@MainActivity, "Incorrect code", Toast.LENGTH_SHORT).show()
            }
        }

        builder.setNegativeButton("Cancel") { dialog, _ -> dialog.cancel() }

        builder.show()
    }


    private fun hideSystemUI() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            window.setDecorFitsSystemWindows(false)
            window.insetsController?.let {
                it.hide(WindowInsets.Type.navigationBars())
                it.systemBarsBehavior = WindowInsetsController.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            }
        } else {
            @Suppress("DEPRECATION")
            window.decorView.systemUiVisibility = (View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                    or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY)
        }
    }
}